### PR TITLE
Remove `higher` dependency

### PR DESCRIPTION
**Pull request by HamedHemati**
This PR update the LaMAML strategy and removes the `higher`  package dependency (related to #1318 ). The new version performs fast-updates using `torch.func` package which requires `torch >= 2.0.0`. In the new version of PyTorch, `functorch` which is the core library behind stateless model execution is stable:
https://pytorch.org/functorch/stable/functorch.html
that's why I removed the old version.

This strategy provides a working example of a meta-learning based strategy in avalanche that doesn't require any external dependencies. 

The accuracy of the model for split-CIFAR-100 is ~70.2. There is a slight improvement (0.2%) in the accuracy which could be related to the internal buffer used in the new version.

### CODE REVIEW

**Comment by AntonioCarta** [comment]
Code looks good and functorch looks like an improvement over higher. The only problem is that it requires PyTorch 2.0.
Is there any clear advantage to using this new version?

Otherwise I suggest to:
- keep both versions, the old one and the new one
- keep the new version private for now (e.g. rename as `_la_maml.py`) so that users use the old one by default
- once we decide to drop pytorch < 2.0 or deprecate higher, switch to the new version

The switch to 2.0 doesn't look problematic but it seems a bit early to enforce it.

**Comment by HamedHemati** [comment]
> * keep the new version private for now (e.g. rename as `_la_maml.py`) so that users use the old one by default

I re-added the `higher` version and renamed the `functorch` version to `lamaml_v2.py`.

**Comment by AntonioCarta** [comment]
Great! can you fix the tests so that I can merge it?

**Comment by HamedHemati** [comment]
The error is caused by package import:
```
ModuleNotFoundError: No module named 'higher'
```

It requires `avalanche[extra]`.

**Comment by HamedHemati** [comment]
> Great! can you fix the tests so that I can merge it?

The error is caused by package import:
```
ModuleNotFoundError: No module named 'higher'
```

It requires `avalanche[extra]`.

File: avalanche/training/supervised/__init__.py [hunk 1]
```diff
@@ -12,3 +12,4 @@
 from .er_ace import ER_ACE, OnlineER_ACE
 from .der import DER
 from .l2p import LearningToPrompt
+from .lamaml import LaMAML
```
Comment (AntonioCarta) [inline avalanche/training/supervised/__init__.py:15]: Remove to fix the error?

**Comment by AntonioCarta** [comment]
I know, but it was introduced in this PR, I believe because of the import in the `__init__`

**Comment by HamedHemati** [comment]
> I know, but it was introduced in this PR, I believe because of the import in the `__init__`

Ah, right. I added it in this PR. Will remove it, thanks.

**Pull request merged**